src/sap/fhir/model/r4/FHIRUtils.js

```diff
@@ -717,6 +717,48 @@ sap.ui.define([
 		return sFullUrl;
 	};
 
+	/**
+	* Extracts resource IDs from the error description in FHIR OperationOutcome.
+	* @param {Array<object>} aOperationOutcome - The array of FHIR OperationOutcome objects.
+	* @returns {Array<string>} - An array containing the extracted resource IDs.
+	* @private
+	* @since 2.4.0
+	*/
+	FHIRUtils.getIdFromOperationOutcome = function (aOperationOutcome) {
+		var aID = [];
+		var text;
+		for (var key in aOperationOutcome) {
+			if (aOperationOutcome.hasOwnProperty(key)) {
+				var operationOutcome = aOperationOutcome[key];
+				var issue = operationOutcome._aIssue[0];
+				if (issue) {
```

Comment:
instead of 3 ifs .. this can be rewritten
`if (issue) {
					text = issue.details.text;
					var aMatchedId = text.match(/[0-9a-fA-F]{8}(?:-[0-9a-fA-F]{4}){3}-[0-9a-fA-F]{12}/);
					aMatchedId && aMatchedId.length > 0 ? aId.push(aMatchedId[0]): [];
				}`

Reply:
With this one if can be excluded but we will have to put an if for text and issue